src/centralstore/incremental/NodeManager.cpp

```diff
@@ -0,0 +1,254 @@
+}
+
+/**
+ * @Deprecated use NodeBlock.get() instead
+ **/
+NodeBlock *NodeManager::get(std::string nodeId) {
+    NodeBlock *nodeBlockPointer = NULL;
+    int debug = std::stoi(nodeId);
+    if (this->nodeIndex.find(nodeId) == this->nodeIndex.end()) {  // Not found
+        return nodeBlockPointer;
+    }
+    unsigned int nodeIndex = this->nodeIndex[nodeId];
+    const unsigned int blockAddress = nodeIndex * NodeBlock::BLOCK_SIZE;
+    NodeBlock::nodesDB->seekg(blockAddress);
+    unsigned int edgeRef;
+    unsigned int propRef;
+    char usageBlock;
+    char label[NodeBlock::LABEL_SIZE];
+
+    if (!NodeBlock::nodesDB->get(usageBlock)) {
+        node_manager_logger.error("Error while reading usage data from block " + std::to_string(blockAddress));
+    }
+
+    if (!NodeBlock::nodesDB->read(reinterpret_cast<char *>(&edgeRef), sizeof(unsigned int))) {
+        node_manager_logger.error("Error while reading edge reference data from block " + std::to_string(blockAddress));
+    }
+
+    if (!NodeBlock::nodesDB->read(reinterpret_cast<char *>(&propRef), sizeof(unsigned int))) {
+        node_manager_logger.error("Error while reading prop reference data from block " + std::to_string(blockAddress));
+    }
+
+    if (!NodeBlock::nodesDB->read(&label[0], NodeBlock::LABEL_SIZE)) {
+        node_manager_logger.error("Error while reading label data from block " + std::to_string(blockAddress));
+    }
+    bool usage = usageBlock == '\1';
+    node_manager_logger.debug("Label = " + std::string(label));
+    node_manager_logger.debug("Length of label = " + std::to_string(strlen(label)));
+    node_manager_logger.debug("DEBUG: raw edgeRef from DB (disk) " + std::to_string(edgeRef));
+
+    nodeBlockPointer = new NodeBlock(nodeId, blockAddress, propRef, edgeRef, label, usage);
+
+    node_manager_logger.debug("DEBUG: nodeBlockPointer after creating the object edgeRef" +
+                              std::to_string(nodeBlockPointer->edgeRef));
+
+    if (nodeBlockPointer->edgeRef % RelationBlock::BLOCK_SIZE != 0) {
+        throw "Exception: Invalid edge reference address = " + nodeBlockPointer->edgeRef;
+    }
+    return nodeBlockPointer;
+}
+
+void NodeManager::persistNodeIndex() {
+    std::ofstream index_db(this->index_db_loc, std::ios::trunc | std::ios::binary);
+    if (index_db.is_open()) {
+        for (auto nodeMap : this->nodeIndex) {
+            char nodeIDC[NodeManager::INDEX_KEY_SIZE] = {0};  // Initialize with null chars
+            std::strcpy(nodeIDC, nodeMap.first.c_str());
+            index_db.write(nodeIDC, sizeof(nodeIDC));
+            unsigned int nodeBlockIndex = nodeMap.second;
+            index_db.write(reinterpret_cast<char *>(&(nodeBlockIndex)), sizeof(unsigned int));
+            node_manager_logger.debug("Writing node index --> Node key = " + std::string(nodeIDC) + " value " +
+                                      std::to_string(nodeBlockIndex));
+        }
+    }
+    index_db.close();
+}
+
+/**
```

Comment:
Can we write function level comments for all of the remaining functions in this PR? Having function level comments is best practice although we have not enforced it in this project so far.

Reply:
Fixed in https://github.com/miyurud/jasminegraph/pull/115/commits/ed31ba0af33e9629173808960eef3cfe90c5dbe9

Reply:
Done